src/core/response_validator.py

```diff
@@ -0,0 +1,739 @@
+"""
+Response Validation and Cleanup for LLM Metric Summaries
+
+This module provides comprehensive validation and cleanup logic to ensure LLM responses 
+contain only the required content without extra explanations, notes, or repetitive sections.
+
+KEY VALIDATION STRATEGIES:
+=========================
+
+1. **Paragraph-based Analysis**: Splits responses into logical paragraphs and validates
+   the expected structure based on response type (OpenShift 4-question vs vLLM 5-requirement).
+
+2. **Content Truncation**: Removes extra content that appears after the required sections,
+   such as "Note:", "Feel free to ask", "Hope this helps", etc.
+
+3. **Bullet Point Validation**: For vLLM responses, ensures the Alerting section contains
+   only bullet points (max 3) and properly handles formatted bullet content.
+
+4. **Sentence Completion**: Removes incomplete sentences that were cut off during truncation
+   while preserving valid bullet points that don't end with punctuation.
+
+5. **Whitespace Normalization**: Cleans up excessive blank lines and trailing whitespace
+   while preserving paragraph structure.
+
+RESPONSE TYPES SUPPORTED:
+========================
+
+- **OPENSHIFT_ANALYSIS**: 4-question format for OpenShift metrics analysis
+- **VLLM_ANALYSIS**: 5-requirement format for vLLM performance analysis  
+- **GENERAL_CHAT**: Free-form responses without structural constraints
+
+VALIDATION FLOW:
+===============
+
+1. Parse response into paragraphs
+2. Identify substantive content paragraphs (filter out headers/questions)
+3. Apply type-specific validation rules
+4. Find optimal truncation point
+5. Clean up whitespace and incomplete sentences
+6. Return cleaned response with validation metadata
+
+This approach ensures consistent, clean responses while preserving all required content.
+"""
+
+import re
+from typing import Dict, List, Tuple, Optional
+from enum import Enum
+
+
+class ResponseType(Enum):
+    """Types of metric analysis responses"""
+    OPENSHIFT_ANALYSIS = "openshift_analysis"  # 4-question format
+    VLLM_ANALYSIS = "vllm_analysis"  # 5-requirement format
+    GENERAL_CHAT = "general_chat"  # Free-form responses
+
+
+class ResponseValidator:
+    """
+    Validates and cleans up LLM responses for metric analysis.
+    
+    This class implements a sophisticated validation system that handles different
+    response formats while removing unwanted extra content that LLMs often add.
+    """
+    
+
+    
+    # Regex patterns for the 4 required OpenShift analysis questions.
+    # These patterns are flexible to handle variations in wording and formatting.
+    OPENSHIFT_QUESTIONS = [
+        r"what'?s?\s+the\s+current\s+.*?state",     # Q1: Current state
```

Comment:
Are we matching `what's` only? What about `what is`? Is `what is` even a valid question? If not, then ignore this comment.